src/envoy/utils/utils.cc

```diff
@@ -34,6 +34,29 @@ const std::string kPerHostMetadataKey("istio");
 // Attribute field for per-host data override
 const std::string kMetadataDestinationUID("uid");
 
+bool getCertSAN(const Network::Connection* connection, bool peer,
+                std::string* principal) {
+  if (connection) {
+    Ssl::ConnectionInfo* ssl =
+        const_cast<Ssl::ConnectionInfo*>(connection->ssl());
```

Comment:
Why const_cast? Could you add a comment here?

Reply:
That's the old code I'm not sure the intention, removed the const_cast in this PR, thanks for catching this.